servers/src/common/adapters.rs

```diff
@@ -647,8 +643,10 @@ impl ChainAdapter for ChainToPoolAndNetAdapter {
 			let cb: CompactBlock = b.clone().into();
 			self.peers().broadcast_compact_block(&cb);
 		} else {
-			// "header first" propagation if we are not the originator of this block
-			self.peers().broadcast_header(&b.header);
+			if self.sync_state.is_synced_once() {
```

Comment:
Still not convinced the added complexity of `is_synced_once` is worth it. There is no difference between the 1st and 2nd sync for a node.
Can we not just use the existing `is_syncing()` here?

Reply:
If using existing `is_syncing`, the fraudulent most work peer will make the gossip block headers propagation pause (because node is in syncing state), until the node leave syncing mode by a fraud detection and banning.

Reply:
🤦‍♂️ Of course, forgot about that.

We should just _always_ broadcast out "header first" here, regardless of syncing or not, regardless of synced_first or not. 
Keep it simple and let peers handle these redundant headers as necessary.

Reply:
I worry there will be too much redundant headers then, and not sure whether the current grin version can handle these flood redundant headers, we don't have much time to test it since only less 4 days before launching.

Also here's a solution to kill that fraudulent work case, I don't understand why simple design is more important than defending from attack.

BTW, does this `is_synced_once` indeed make design become a very complex one?

Reply:
Look like we couldn't agree each other between us about whether `always` broadcast out "header first" including initial syncing stage, perhaps it's good to invite more pair of eyes on this review 😄   @ignopeverell  what's your opinion on this?